pad_api_data/pad_etl/processor/enemy_skillset.py

```diff
@@ -167,272 +137,22 @@ def positions_2d_bitmap(bits_arr):
-            return 'Delay active skills by {:d} turns'.format(min_turns)
-        else:
-            return 'Delay active skills by {:d}~{:d} turns'.format(min_turns, max_turns)
-
-    @staticmethod
-    def orb_lock(count, attributes):
-        if count == 42:
-            return 'Lock all {:s} orbs'.format(', '.join(attributes))
-        else:
-            return 'Lock {:d} random {:s} orbs'.format(count, ', '.join(attributes))
-
-    @staticmethod
-    def orb_seal(turns, position_type, positions):
-        return 'Seal {:s} {:s} for {:d} turns'.format(', '.join([ordinal(x) for x in positions]), position_type, turns)
-
-    @staticmethod
-    def cloud(turns, width, height, x, y):
-        if width == 6 and height == 1:
-            shape = 'Row of'
-        elif width == 1 and height == 5:
-            shape = 'Column of'
-        else:
-            shape = '{:d}x{:d}'.format(width, height)
-        pos = []
-        if x is not None and shape != 'Row of':
-            pos.append('{:s} row'.format(ordinal(x)))
-        if y is not None and shape != 'Column of':
-            pos.append('{:s} column'.format(ordinal(y)))
-        if len(pos) == 0:
-            pos.append('random location')
-        return '{:s} cloud appear for {:d} turns at {:s}'.format(shape, turns, ', '.join(pos))
-
-    @staticmethod
-    def turn_change(turn_counter, threshold=None):
-        if threshold:
-            return 'Enemy turn counter change to {:d} when HP <= {:d}%'.format(turn_counter, threshold)
-        else:
-            return 'Enemy turn counter change to {:d}'.format(turn_counter)
-
-    @staticmethod
-    def attribute_block(turns, attributes):
-        return 'Unable to match {:s} orbs for {:d} turns'.format(', '.join(attributes), turns)
-
-    @staticmethod
-    def spinners(turns, speed, position_description):
-        return '{:s} orbs change every {:.1f}s for {:d} turns'.format(position_description, speed / 100, turns)
-
-    @staticmethod
-    def max_hp_change(turns, max_hp, change_type):
-        if change_type == 'flat':
-            return 'Change player HP to {:d} for {:d} turns'.format(max_hp, turns)
-        else:
-            return 'Change player HP to {:d}% for {:d} turns'.format(max_hp, turns)
-
-    @staticmethod
-    def death_cry(message):
-        if message is None:
-            return 'Show death effect'
-        else:
-            return 'Show message: {:s}'.format(message)
-
-    @staticmethod
-    def attribute_exists(atts):
-        return 'when {:s} orbs are on the board'.format(', '.join(atts))
-
-    @staticmethod
-    def countdown(counter):
-        return 'Display \'{:d}\' and skip turn'.format(counter)
-
-
 # Condition subclass
 
 class ESCondition(object):
-    def __init__(self, ai, rnd, params_arr, description=None):
+    def full_description(self):
+        return Describe_EN.condition(max(ai, rnd), self.hp_threshold, self.forced_one_time is not None, self.extra_description)
```

Comment:
max(ai, rnd) should be self.ai self.rnd right?

Might be simpler to just pass the whole condition in though right. I guess you're differentiating between description and full_description hmm. Could have desc.condition_full delegate to desc.condition with the extra logic listed here.